MaterialDesignThemes.Wpf/SnackbarMessageQueue.cs

```diff
@@ -152,11 +136,12 @@ public SnackbarMessageQueue()
         public SnackbarMessageQueue(TimeSpan messageDuration)
         {
             _messageDuration = messageDuration;
-            Task.Factory.StartNew(PumpAsync, TaskCreationOptions.LongRunning);
+            _dispatcher = Dispatcher.FromThread(Thread.CurrentThread)
+                          ?? throw new InvalidOperationException("SnackbarMessageQueue must be created in a dispatcher thread");
```

Comment:
Why are we not using Application.Current.Dispatcher?

Reply:
`Application.Current.Dispatcher` is not clear in applications with multiple dispatchers (I have such in my application).

I took this from the `DispatcherObject` where it gets its dispatcher with `Dispatcher.CurrentDispatcher`. But this returns a new Dispatcher of the current does not have one. See [here](https://referencesource.microsoft.com/#windowsbase/Base/System/Windows/Threading/Dispatcher.cs).

If you like I can change it to `Dispatcher.CurrentDispatcher`